docs/articles/Getting-Started.md

```diff
@@ -0,0 +1,386 @@
+two, we can asynchronously load all the data needed for a module and add it to the `store` when
+a `reducer` is supplied to the `holocron` configuration.
+
+`src/components/MyModule.jsx`
+
+```jsx
+import React from 'react';
+import { fromJS } from 'immutable';
+
+function DataVisualizer({ data, loaded }) {
+  return (
+    <div>
+      {/* do something with the data when it loads */}
+    </div>
+  );
+}
+
+export default function MyModule({ moduleState = {}, moduleLoadStatus = 'loading' }) {
+  return (
+    <DataVisualizer data={moduleState} loaded={moduleLoadStatus === 'loaded'} />
+  );
+}
+
+MyModule.holocron = {
+  name: 'my-module',
+  reducer: (moduleState = fromJS({}), action) => {
+    if (action.type === 'my-action') return fromJS(action.data);
+    return moduleState;
+  },
+  loadModuleData: async ({ store: { dispatch }, fetchClient }) => {
+    const response = await fetchClient('url/to/data.json');
+    const data = await response.json();
+    dispatch({ type: 'my-action', data });
+  },
+};
+```
+
+Both `moduleState` and `moduleLoadStatus` are props given to each Holocron module.
+All state is `immutable` and the `reducer` provided will be expected to return
+Immutable compliant types (which includes JavaScript primitives). `loadModuleData`
+is passed the Redux `store`, along with the `fetchClient`, `ownProps` and the
+`module` itself. `fetchClient` can be configured with `appConfig` which will
+learn about in the next section.
+
+> #### More Info
+>
+> **API**
+>
+> [Loading Data](../api/modules/loading-data)
+>
+> [State Management](../api/modules/state-management)
+>
+> **Recipes**
+>
+> [Enabling Server Side Render](../recipes/enabling-serverside-rendering)
+>
+> [Internationalization](../recipes/internationalizing-your-module)
+>
+> ##### Packages
+>
+> [`immutable`](https://immutable-js.github.io/immutable-js/)
+>
+> [`redux`](https://redux.js.org/)
+>
+> [`react-redux`](https://react-redux.js.org/)
+>
+> [`reselect`](https://github.com/reduxjs/reselect)
+
+### Configuring One App
+
+When One App first starts up on the server, it loads in all the modules
+and looks for `Module.appConfig` in each module to configure the app runtime.
+The root module is used to configure many aspects of One App, including the
+state configuration that is available for every module to use when rendering.
+
+`src/appConfig.js`
+
+```js
+export default {
+  csp: "default-src 'self';",
+  providedStateConfig: {
+    theme: {
+      client: 'my-theme-name',
+      server: 'my-theme-name',
+    },
+  },
+};
+```
+
+The `appConfig` property is meant strictly for the server side,
+take advantage of `global.BROWSER` to ensure that the `appConfig`
+is only bundled with the server side build.
+
+`src/components/MyModule.jsx`
+
+```jsx
+import React from 'react';
+import { useSelector } from 'react-redux';
+
+export default function MyModule() {
+  const { theme } = useSelector(
+    (state) => state.get('config'),
```

Comment:
maybe consider this example to load a single config value from the state? for best practices and performance we should recommend avoiding `state.ToJS()` if possible